src/SignalR/common/Shared/MessageBuffer.cs

```diff
@@ -121,15 +121,16 @@ private async Task RunTimer()
 
     public ValueTask<FlushResult> WriteAsync(SerializedHubMessage hubMessage, CancellationToken cancellationToken)
     {
-        return WriteAsyncCore(hubMessage.Message!, hubMessage.GetSerializedMessage(_protocol), cancellationToken);
+        // Default to HubInvocationMessage as that's the only type we use SerializedHubMessage for currently. Should harden this in the future.
+        return WriteAsyncCore(hubMessage.Message?.GetType() ?? typeof(HubInvocationMessage), hubMessage.GetSerializedMessage(_protocol), cancellationToken);
```

Comment:
```suggestion
        return WriteAsyncCore(typeof(HubInvocationMessage), hubMessage.GetSerializedMessage(_protocol), cancellationToken);
```

If we're really confident with what the comment is saying, why not change it to the above? We could add a Debug.Assert that `hubMessage.Message is null || hubMessage.Message.GetType() == typeof(HubInvocationMessage)`.